cmd/juju/controller/listmodels_test.go

```diff
@@ -309,11 +362,19 @@ func (s *ModelsSuite) TestModelsMachineInfo(c *gc.C) {
 		"carlotta/test-model2         dummy         active             0      -  write   2015-03-01\n"+
 		"daiwik@external/test-model3  dummy         destroying         0      -  -       never connected\n"+
 		"\n")
+	s.checkAPICalls(c, "BestAPIVersion", "ListModels", "ModelInfo", "Close")
 }
 
-func (s *ModelsSuite) TestAllModelsWithOneUnauthorised(c *gc.C) {
+// This test is only needed for older api versions as
+// whether the user has an access to a model will be checked on
+// the api side and the model data will not be sent.
+func (s *BaseModelsSuite) TestAllModelsWithOneUnauthorised(c *gc.C) {
```

Comment:
My thinking was - either the test comment is incorrect or this should reside on ModelSuiteV3 only. It looks like the test is passing for both v3 and v4 suites, so maybe get rid of the comment?